app/src/androidTest/java/com/duckduckgo/app/browser/shortcut/ShortcutReceiverTest.kt

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.shortcut
+
+import android.content.Intent
+import com.duckduckgo.app.CoroutineTestRule
+import com.duckduckgo.app.global.events.db.UserEventsStore
+import com.duckduckgo.app.global.useourapp.UseOurAppDetector.Companion.USE_OUR_APP_SHORTCUT_URL
+import com.duckduckgo.app.runBlocking
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.never
+import com.nhaarman.mockitokotlin2.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class ShortcutReceiverTest {
+
+    @get:Rule
+    var coroutinesTestRule = CoroutineTestRule()
+
+    private val mockUserEventsStore: UserEventsStore = mock()
+    private val mockPixel: Pixel = mock()
+    private lateinit var testee: ShortcutReceiver
+
+    @Before
+    fun before() {
+        testee = ShortcutReceiver(mockUserEventsStore, coroutinesTestRule.testDispatcherProvider, mockPixel)
+    }
+
+    @Test
+    fun whenIntentReceivedIfUrlIsFromUseOurAppUrlThenRegisterTimestamp() = coroutinesTestRule.runBlocking {
+        val intent = Intent()
+        intent.putExtra(ShortcutBuilder.SHORTCUT_URL_ARG, USE_OUR_APP_SHORTCUT_URL)
+        intent.putExtra(ShortcutBuilder.SHORTCUT_TITLE_ARG, "Title")
+        testee.onReceive(null, intent)
+
+        verify(mockUserEventsStore).registerUserEvent(any())
+    }
+
+    @Test
+    fun whenIntentReceivedIfUrlIsFromUseOurAppUrlThenFirePixel() {
+        val intent = Intent()
+        intent.putExtra(ShortcutBuilder.SHORTCUT_URL_ARG, USE_OUR_APP_SHORTCUT_URL)
+        intent.putExtra(ShortcutBuilder.SHORTCUT_TITLE_ARG, "Title")
+        testee.onReceive(null, intent)
+
+        verify(mockPixel).fire(Pixel.PixelName.USE_OUR_APP_SHORTCUT_ADDED)
+    }
+
+    @Test
+    fun whenIntentReceivedIfUrlIsNotFromUseOurAppUrlThenDoNotRegisterTimestamp() = coroutinesTestRule.runBlocking {
```

Comment:
Reference to Timestamp should be `Event`

Reply:
Done!